### ISSUE TITLE
`diart.stream` crashes on files with short duration

### ISSUE THREAD

**Issue by ankurdhuriya** [juanmc2005/StreamingSpeakerDiarization#118]
I am getting file not found error while running diart.stream audio_file

**Platform, Operating System, and Programming Language**

**Platform** : Apple M2
**OS** : macOS Ventura 13.1
**Programming Language** : Python 3.8.15

```
StreamingSpeakerDiarization % diart.stream ~/Downloads/test.wav 
Streaming test: 0chunk [00:00, ?chunk/s]Traceback (most recent call last):
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 55, in _on_error
    on_error(exception)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/observer.py", line 43, in on_error
    self._on_error_core(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/observer.py", line 49, in _on_error_core
    self._handler_on_error(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/internal/basic.py", line 34, in default_error
    raise err
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 66, in _on_completed
    on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/sinks.py", line 53, in on_completed
    self.patch()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/sinks.py", line 36, in patch
    annotations = list(load_rttm(self.path).values())
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pyannote/database/util.py", line 306, in load_rttm
    data = pd.read_csv(
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1735, in _make_engine
    self.handles = get_handle(
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/common.py", line 856, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: '/Users/ankurdhuriya/Downloads/test.rttm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/bin/diart.stream", line 8, in <module>
    sys.exit(run())
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/stream.py", line 71, in run
    inference()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/inference.py", line 200, in __call__
    self.source.read()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/sources.py", line 112, in read
    self.stream.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/subject/subject.py", line 89, in on_completed
    super().on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/observer.py", line 56, in on_completed
    self._on_completed_core()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/subject/subject.py", line 97, in _on_completed_core
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  [Previous line repeated 3 more times]
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/windowwithcount.py", line 75, in on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/merge.py", line 119, in on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 63, in _on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 63, in _on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/merge.py", line 119, in on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 70, in _on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 63, in _on_completed
    observer.on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 44, in on_completed
    self._on_completed()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 68, in _on_completed
    observer.on_error(e)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 34, in on_error
    self._on_error(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 34, in on_error
    self._on_error(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 34, in on_error
    self._on_error(error)
  [Previous line repeated 1 more time]
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/operators/do.py", line 57, in _on_error
    observer.on_error(e)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/rx/core/observer/autodetachobserver.py", line 34, in on_error
    self._on_error(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/inference.py", line 157, in _handle_error
    sink.on_error(error)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/sinks.py", line 50, in on_error
    self.patch()
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/diart/sinks.py", line 36, in patch
    annotations = list(load_rttm(self.path).values())
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pyannote/database/util.py", line 306, in load_rttm
    data = pd.read_csv(
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1735, in _make_engine
    self.handles = get_handle(
  File "/Users/ankurdhuriya/opt/miniconda3/envs/pyannote/lib/python3.8/site-packages/pandas/io/common.py", line 856, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: '/Users/ankurdhuriya/Downloads/test.rttm'
Streaming test: 0chunk [00:00, ?chunk/s]

```

**Comment by juanmc2005** [comment]
Hi @ankurdhuriya,

It looks like the your audio stream is ending without receiving any predictions. Could you provide more information about your `test.wav` file?

**Comment by ankurdhuriya** [comment]
@juanmc2005 
this is my audio file
Duration: 00:00:01.28, bitrate: 256 kb/s
 Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 16000 Hz, 1 channels, s16, 256 kb/s

**Comment by juanmc2005** [comment]
Yes the problem is definitely the duration. The segmentation model is expecting 5s inputs and it's not being able to fill the audio buffer even once.

It's due to [this line](https://github.com/juanmc2005/StreamingSpeakerDiarization/blob/main/src/diart/operators.py#L84) in `rearrange_audio_stream`.
For a permanent fix, I see two solutions:

1) Add zero padding to the audio file
2) Capture the event and raise a meaningful error

I think a good solution would be to add automatic zero padding when the file is shorter than `config.duration` and print a warning so the user is aware of the problem.

Could you first add the zero padding to check if it solves the issue? I'll add this to the backlog for version 0.7.

**Comment by juanmc2005** [comment]
Implemented in PR #135
